shared/subspace-data-retrieval/src/piece_getter.rs

```diff
@@ -56,6 +83,18 @@ impl ObjectPieceGetter for NewArchivedSegment {
 
         Ok(None)
     }
+
+    async fn get_pieces<'a, PieceIndices>(
```

Comment:
Is this actually desirable though? I removed default implementation after transition from other places and implemented it explicitly where it was still necessary.

Reply:
Are you suggesting having separate traits for getting a single piece, and getting multiple pieces?
I’m happy to do that.